influxql/parser.go

```diff
@@ -577,6 +577,15 @@ func (p *Parser) parseSelectStatement(tr targetRequirement) (*SelectStatement, e
 		return nil, err
 	}
 
+	// Set if the query is a raw data query or one with an aggregate
+	stmt.IsRawQuery = true
```

Comment:
No, the second example will be a single statement. The `min` aggregate
function will basically pass different things to the reducer based on the
other parts of the query. When the `min` mapper and reducer functions are
initialized, they'll take arguments for tags and fields that should be
included.

Those will then be processed by the processor that handles the output from
the reducer. It's a bit tricky, but the identification of the other fields
needs to happen within the query scan that handles the min.

On Thu, Mar 19, 2015 at 5:02 PM, otoolep <notifications@github.com> wrote:

> In influxql/parser.go
> <https://github.com/influxdb/influxdb/pull/2004#discussion_r26797273>:
>
> > @@ -577,6 +577,15 @@ func (p *Parser) parseSelectStatement(tr targetRequirement) (*SelectStatement, e
> >  		return nil, err
> >  	}
> >
> > +	// Set if the query is a raw data query or one with an aggregate
> > +	stmt.IsRawQuery = true
>
> OK, but doesn't your second example still result in two substatements
> within the code? One for host and one for min(value). It's been a while
> since I looked at the relevant code, I just recall that "substatement"
> concept and want to check that it's all OK.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/2004/files#r26797273>.
>